### ISSUE TITLE
Clicking on MSCZ file fails to open if Mu3.7 is already running

### ISSUE THREAD

**Issue by BanjoJake** [Jojo-Schmitz/MuseScore#679]
I click on an MSCZ file, and it launches Musescore and opens the file. I click on another, and it doesn't.
   Or:
I launch Musescore (empty). I click on an MSCZ file, and it does not open.

This happens using File Explorer (and Z-Tree) in Windows 10 & 11
Observed in v3.7.0.11299493561 - ba6d337 and later (64-bit).

3.7.0.11258847346, 009a9bd and earlier work OK

See https://musescore.org/en/node/370307
______________________
Background:  
First, I did this once, months ago: I unzipped the folder inside the 7z (eg, MuseScore-3.7.0.7099354748-x86_64) into G:\APPS, then renamed it "MuseScore-3.7.0". Then I right-clicked an mscz file, then "Open with" then "Choose another app", then I navigated to G:\APPS\MuseScore-3.7.0\bin\MuseScore3.exe

Now, whenever I want to update MuseScore-3.7, I delete the MuseScore-3.7.0 folder in G:\APPS, unzip the latest folder (eg, MuseScore-3.7.0.11343013698-x86_64) into G:\APPS, then rename it to "MuseScore-3.7.0".

**Comment by Johan-v** [comment]
This is so weird.
I hope you can fix it.
It is one of the reasons I don't use MU4 in my classroom. Fast loading and switching scores and.
My last version I can use is Mu3.7_242831609_Win64_3

**Comment by Jojo-Schmitz** [comment]
File > Open should always work though

**Comment by Johan-v** [comment]
I know but to slow and not compatible with my workflow.

I need very fast access to different files. (musescore, pdf, word, audio images etc.)
Those are in many many different folders for each week and lesson and class.

For navigating I use a link in excel to that folder for eacht lesson.
That link jumps in a second to the folder in file explorer.
There I can start every file I need immediately without navigating and losing time or searching.
Everything is instant.

**Comment by BanjoJake** [comment]
Not to intrude on the dev's territory, but this looks like it may have some relevance:
  stackoverflow.com/questions/424368/opening-a-known-file-type-into-running-instance-of-custom-app-net

**Comment by Johan-v** [comment]
Any news about this? 
Thanks

**Comment by Jojo-Schmitz** [comment]
none yet

**Comment by BanjoJake** [comment]
Hi, sorry, don't mean to be a pest, just an FYI, I'm stuck on 3.7.0.11258847346 - 009a9bd since this bug came up and not able to try any of the new updates.   But I certainly respect your priorities.  Many thanks for all you do.

**Comment by Jojo-Schmitz** [comment]
File > Open, as well as dragging a score file into the 3.7 windows should open it though

**Comment by BanjoJake** [comment]
> File > Open, as well as dragging a score file into the 3.7 window should open it though.

True.  So to get into the weeds:  As a volunteer, I maintain large repertoires for four community bands, and use Z-tree (www.ztree.com) to maintain them; it's more precise and much faster to use than Windows File Explorer.  For one example, Explorer makes it easy (for me anyway) to accidentally move files trying to use drag&drop.    Z-Tree will display all the files in a single list across all drives and subfolders, sorts and applies filename filters instantly.    I use it for PDFs, JPGs, ABCs, XMLs, SIBs, as well as MSCZs.   There are hundreds of scores (including transposed and different clefs) scattered among several folders and drives.  Once I scroll down to a file I need, typing "o" opens it into Musescore.   Basically it's impossible for me to work at anything close to this level of efficiency with the latest Mu3.7 versions.   

Thanks for understanding, and once again for the tremendous work you're doing !

**Comment by Johan-v** [comment]
This is so strange.
What is going on here?
What can be the logic behing this?
Somebody has a clue?

**Comment by Jojo-Schmitz** [comment]
I certainly have no clue, I do't recall any chage that might have caused this

**Comment by BanjoJake** [comment]
Well, we know it started with Commit ba6d337, miiizen
 authored and Jojo-Schmitz committed on Oct 11

Description is:
"Remove the NoteLine element
Backport of https://github.com/musescore/MuseScore/pull/24993
Also fix most clazy warnings" (sic)

Since it's unlikely that a high-level programming change would affect an OS-level function, I wonder if maybe miiizen compiled using some different libraries or DLLs ?  Or maybe there was some silent update to the C++ (or whatever) compiler around that time ?

**Comment by Jojo-Schmitz** [comment]
Let's test whether that cimmit really is the culprit by reverting it, see PT #730

**Comment by Jojo-Schmitz** [comment]
OK, next step then...

**Comment by Johan-v** [comment]
This one is working for me.
Mu3.7_243541004_Win64_730_Fix GH#679_ Clicking on MSCZ file fails to open if Mu3.7 is already running

**Comment by Jojo-Schmitz** [comment]
Ok, so now I need to find which of these crazy warnings fixes is the culprit

**Comment by BanjoJake** [comment]
Sorry about my ignorance of the dev process here, but maybe ask miiizen if he has any ideas ?

**Comment by BanjoJake** [comment]
Yes !   MuseScore (64-bit): 3.7.0.12417897231/ 450fbab works on Windows 11
Will test now on Windows 10

**Comment by Jojo-Schmitz** [comment]
OK, now only my changes to musescore.cpp remain to get checked...

**Comment by BanjoJake** [comment]
Also tested on Windows 10, and 32-bit (Win10 & 11).  Confirmed all working properly (v3.7.0.12417897231 / 450fbab).

**Comment by Jojo-Schmitz** [comment]
Seems I can now reproduce the issue, so finding the culprit change should be much easier

**Comment by Jojo-Schmitz** [comment]
A final test please...

**Comment by BanjoJake** [comment]
Confirmed MuseScore (64-bit): 3.7.0.12431681744 / 561764a works correctly on Windows 10 64-bit !

**Comment by Johan-v** [comment]
Top! Thank you.

Op vr 20 dec. 2024 14:14 schreef Joachim Schmitz ***@***.***>:

**Comment by Jojo-Schmitz** [comment]
See https://musescore.org/en/node/377907
